andela/ah-shakas
# #160577483 Liking/Disliking Articles Functionality

#### What does this PR do?
Make user's of Author's Haven able to like/dislike and delete a like/dislike to an article.
#### Description of Task to be completed?
Have the following endpoints working
```POST /api/articles/<slug>/like/``` - post a like/dislike
```DELETE /api/articles/<slug>/like/``` - delete a like/dislike
#### How should this be manually tested?
- Clone [this](https://github.com/andela/ah-shakas.git) repository and cd into the folder.
- Run the command ```git checkout ft-liking-unliking-articles-160577483```
- Create a ```.env``` file with the following values
```
DB_NAME=
DB_HOST=
DB_PASSWORD=
DB_USER=
EMAIL_HOST=smtp.sendgrid.net
EMAIL_HOST_USER=
EMAIL_HOST_PASSWORD=
EMAIL_PORT=
```
- Install the project dependencies by running ```pipenv install```
- Then to run the tests ren the command ```python manage.py test```
- Using [Postman](https://www.getpostman.com/) test every endpoint above after running the command ```python manage.py runserver```
- Create a user account by accessing ```POST api/users/``` and sending the following payload ```{ "user":{ "username":"username", "email":"username@gmail.com", "password": "password1234"}}```
- Confirm signup from the email sent to the address posted in the payload.
- Then login to get token to perform manual tests
_key_: Content-Type value: application/json
_key_: Authorization value: JWT Token generated either on signup or login
- Create an article by accessing ```POST api/articles/``` and pass in the following payload ```{"article":{"title":"title", "Title", "body": "My article"}}```
- Test the endpoins above by passing the following payload ```{"likes":true}```
#### Any background context you want to provide?
Author's Haven application has three article states when it comes to likes. A reader can have no response to an article, and either likes or dislike an article. One can exist in only one of these states.
#### What are the relevant pivotal tracker stories?
[#160577483 Users should be able to like and or dislike articles](https://www.pivotaltracker.com/n/projects/2198533)
#### Screenshots (if appropriate)
<img width="906" alt="screen shot 2018-10-16 at 11 07 31" src="https://user-images.githubusercontent.com/28476798/47001885-31def500-d134-11e8-9626-10efbb6e17c8.png">
<img width="908" alt="screen shot 2018-10-16 at 11 07 57" src="https://user-images.githubusercontent.com/28476798/47001898-3b685d00-d134-11e8-89f1-3fe0002f0844.png">

<img width="907" alt="screen shot 2018-10-16 at 11 09 06" src="https://user-images.githubusercontent.com/28476798/47001915-43c09800-d134-11e8-9cc0-6714e94e34b9.png">

<img width="909" alt="screen shot 2018-10-16 at 11 09 39" src="https://user-images.githubusercontent.com/28476798/47002050-9306c880-d134-11e8-8aa2-949501a1e194.png">

<img width="908" alt="screen shot 2018-10-16 at 11 10 10" src="https://user-images.githubusercontent.com/28476798/47001986-6eaaec00-d134-11e8-9715-a1c3f8614eeb.png">
